src/Stratis.Bitcoin/Base/ChainRepository.cs

```diff
@@ -43,22 +43,27 @@ public Task LoadAsync(ConcurrentChain chain)
             {
                 using (DBreeze.Transactions.Transaction transaction = this.dbreeze.GetTransaction())
                 {
+                    transaction.ValuesLazyLoadingIsOn = false;
                     ChainedBlock tip = null;
-                    bool first = true;
+                    Row<int, BlockHeader> firstRow = transaction.Select<int, BlockHeader>("Chain", 0);
+                    if (!firstRow.Exists)
+                        throw new InvalidOperationException("Block header with index 0 doesn't exist in the database");
+                    BlockHeader previousHeader = firstRow.Value;
+                    Guard.Assert(previousHeader.GetHash() == chain.Genesis.HashBlock); // can't swap networks
```

Comment:
@Aprogiena are you happy with this?

Reply:
What happens if this is a new fresh node will we still have raw zero?

Reply:
of course not, one does not just make me happy, you know that ...

but making it better is outside of the scope of this PR, that's like most of the changes we make now - we make incremental changes that improve the code somehow, but don't make it perfect, so it can still be improved later

so in absolute terms - no, i am not happy with this
in relative terms - yes, for this PR, this is good enough

there are two problems in this code
- using assert for user error instead of handling gracefully (e.g. via throw and catch at higher level)
- there is a penalty from rare user error case to 99.99 % of runs, which is wrong by principle - i'd rather place the check AFTER the chain is processed thus saving that one GetHash call, which would save that GetHash call for those 99.99 % of runs and make that 1 faulty run in 10000 wait for the whole chain loading; so the principle is that such a rare case should not affect the major code flow, which is not satisfied here

but again - for this PR, good enough

Reply:
>What happens if this is a new fresh node will we still have raw zero?

It would throw the exception "Block header with index 0 doesn't exist in the database". 
Looks like it wasn't a good idea, fixed in the latest commit.